Cluster Operator dynamic logging configuration (#3328)
strimzi/strimzi-kafka-operator: documentation/modules/ref-operator-cluster.adoc

+
If you are using a different DNS domain name suffix in your cluster, change the `KUBERNETES_SERVICE_DNS_DOMAIN` environment variable from the default to the one you are using in order to establish a connection with the Kafka brokers.

[discrete]
== Logging configuration

A `ConfigMap` containing logging configuration is created when installing the Cluster Operator.

> This `ConfigMap` is described in the file `install/cluster-operator/050-ConfigMap-strimzi-cluster-operator-logging.yaml`.

Reviewer: I would avoid using "logging" in the name tbh. We don't know if at some point we will need to add more into this `ConfigMap` and having "logging" in the name could be misleading. Wdyt?

You configure Cluster Operator logging by changing the `ConfigMap`.

> To update the logging configuration, you can edit the `050-ConfigMap-strimzi-cluster-operator-logging.yaml` file and then run the following command:

Reviewer: I think you should mention that the logging configuration is in the `log4j2.properties` data field.
If in the future we will use the `ConfigMap` for something more, we will have more data fields.

+
If you are using a different DNS domain name suffix in your cluster, change the `KUBERNETES_SERVICE_DNS_DOMAIN` environment variable from the default to the one you are using in order to establish a connection with the Kafka brokers.

> == Logging configuration

Author: @PaulRMellor @laidan6000 
Could you please check these changes?

Reviewer 2: @stanlyDoge :+1:

> Together with Cluster Operator, `ConfigMap` containing its logging configuration is created.

Reviewer 3: Thsi would sound better to me personally at least.

```suggestion
When installing Cluster Operator, `ConfigMap` containing its logging configuration is created.
```

This `ConfigMap` is described in the file `install/cluster-operator/050-ConfigMap-strimzi-cluster-operator-logging.yaml`.
Cluster Operator logging is then configured by changing this `ConfigMap`.

You can change the file and then use:
[source,shell,subs=+quotes]
kubectl apply -f _install/cluster-operator/050-ConfigMap-strimzi-cluster-operator-logging.yaml_

Or you can edit the `ConfigMap` by:
[source,shell,subs=+quotes]
kubectl edit cm strimzi-cluster-operator-logging

After the changes are done in the `ConfigMap`, the logging configuration is reloaded in time interval of specified `monitorInterval` seconds.

> NOTE: Created `ConfigMap` contains entry `monitorInterval`.

Reviewer 4: Also, I suggest moving the note to the end of this section, i.e. line 145.

> Please ensure yourself you are not removing this entry as it serves for dynamic reloading of logging configuration.

Reviewer 2: ```suggestion
Check you are not removing this entry as it serves for dynamic reloading of logging configuration.
```

Author: Removed in https://github.com/strimzi/strimzi-kafka-operator/pull/3328#discussion_r460786285

> Entry `monitorInterval` configures frequency of logging reloading in the seconds (see link:https://logging.apache.org/log4j/2.x/manual/configuration.html#Automatic_Reconfiguration[Log4j2 Automatic Reconfiguration^]).

Reviewer 2: ```suggestion
Use `monitorInterval` to configure the frequency of logging reloading in seconds (see link:https://logging.apache.org/log4j/2.x/manual/configuration.html#Automatic_Reconfiguration[Log4j2 Automatic Reconfiguration^]).
```

Author: Removed in https://github.com/strimzi/strimzi-kafka-operator/pull/3328#discussion_r460786285

> If the `ConfigMap` is not present during deploying of Cluster Operator, default logging values are used.

Reviewer 4: ```suggestion
If the `ConfigMap` is missing when the Cluster Operator is deployed, the default logging values are used.
```

> If the `ConfigMap` is deleted by an accident after the Cluster Operator was deployed, the last loaded configuration is used.

Reviewer 4: ```suggestion
If the `ConfigMap` is accidentally deleted after the Cluster Operator is deployed, the most recently loaded logging configuration is used.
```

> After `ConfigMap` recreation, the new configuration is loaded.

Reviewer 3: Maybe better this way?

```suggestion
After recreating the `ConfigMap`, the new configuration is loaded.
```

> After recreating the `ConfigMap`, the new configuration is loaded.

Reviewer 4: ```suggestion
Create a new `ConfigMap` to load a new logging configuration.
```

+
If you are using a different DNS domain name suffix in your cluster, change the `KUBERNETES_SERVICE_DNS_DOMAIN` environment variable from the default to the one you are using in order to establish a connection with the Kafka brokers.

[discrete]
== Configuration by ConfigMap

> Currently only Cluster Operator logging can be configured by ConfigMap.

Reviewer 5: ```suggestion
The Cluster Operator's logging is configured by the XXX_CM_name_XXX ConfigMap.
```

A `ConfigMap` containing logging configuration is created when installing the Cluster Operator.
This `ConfigMap` is described in the file `install/cluster-operator/050-ConfigMap-strimzi-cluster-operator.yaml`.
You configure Cluster Operator logging by changing the data field `log4j2.properties` in this `ConfigMap`.

To update the logging configuration, you can edit the `050-ConfigMap-strimzi-cluster-operator.yaml` file and then run the following command:
[source,shell,subs=+quotes]
kubectl apply -f _install/cluster-operator/050-ConfigMap-strimzi-cluster-operator.yaml_

Alternatively, edit the `ConfigMap` directly:
[source,shell,subs=+quotes]
kubectl edit cm strimzi-cluster-operator

> To change the frequency of the reload interval, set a time in seconds in the monitorInterval option in the created ConfigMap.

Reviewer 5: ```suggestion
To change the frequency of the reload interval, set a time in seconds in the `monitorInterval` option in the created `ConfigMap`.
```
